lib/utils/houchdb.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const PouchW = require('pouchdb-wrapper');
+const _ = require('lodash');
+const defaultOptions = {
+    namespace: 'default',
+    port: 5984,
+    hostname: 'localhost',
+    protocol: 'http',
+    db: 'my-db',
+    plugins: []
+};
+
+const register = function(server, options, next) {
+    options = _.merge(defaultOptions, options);
+
+    // register PouchDB plugins
+    // this appears safe to do repeatedly, every time we call this function
+    _.forEach(options.plugins, (plugin) => {
+        PouchW.PouchDB.plugin(plugin);
+    });
+
+    let config;
+    if (options.path) {
+        config = {
+            path: options.path,
+            name: options.name
+        };
+    } else if (options.conn) {
+        config = {
+            name: options.name,
+            conn: options.conn
+        };
+    }
+
+    const db = new PouchW(config);
+    const pluginName = module.exports.attributes.name;
+    const exposableObj = {};
+    exposableObj[options.name] = db;
+    exposableObj.constructor = PouchW;
+
+    // throw error if namespace already exists
+    if (
+        server.plugins[pluginName] &&
+        server.plugins[pluginName][options.name]
+    ) {
+        const msg = 'PouchDB namespace `' + options.prefix + '` already exists';
+        next(new Error(msg));
+    }
+
+    // attempt to get info from the db
+    return db.info()
```

Comment:
That is the desired functionality in this case. We want the server to fail to start if the remote database is unreachable or does not exist. 

In cases where the remote store needs to be created (first time the server is started?), we'll need a way to wait for the couch store to be generated. Any ideas? Perhaps perform a `get()` operation first?